runtime/fastly/builtins/backend.cpp

```diff
@@ -1441,16 +1479,57 @@ bool Backend::constructor(JSContext *cx, unsigned argc, JS::Value *vp) {
     if (!JS_GetProperty(cx, configuration, "sniHostname", &sni_hostname_val)) {
       return false;
     }
-    auto sni_hostname = JS::ToString(cx, sni_hostname_val);
-    if (!sni_hostname) {
+    if (!Backend::set_sni_hostname(cx, backend, sni_hostname_val)) {
+      return false;
+    }
+  }
+
+  JS::RootedValue client_cert_val(cx);
+  if (!JS_HasProperty(cx, configuration, "clientCertificate", &found)) {
+    return false;
+  }
+  if (found) {
+    if (!JS_GetProperty(cx, configuration, "clientCertificate", &client_cert_val)) {
+      return false;
+    }
+    if (!client_cert_val.isObject()) {
+      JS_ReportErrorNumberASCII(cx, FastlyGetErrorMessage, nullptr,
+                                JSMSG_BACKEND_CLIENT_CERTIFICATE_NOT_OBJECT);
+      return false;
+    }
+    JS::RootedObject client_cert_obj(cx, &client_cert_val.toObject());
+
+    JS::RootedValue client_cert_cert_val(cx);
+    if (!JS_HasProperty(cx, client_cert_obj, "certificate", &found)) {
       return false;
     }
-    if (JS_GetStringLength(sni_hostname) == 0) {
+    if (!found) {
       JS_ReportErrorNumberASCII(cx, FastlyGetErrorMessage, nullptr,
-                                JSMSG_BACKEND_SNI_HOSTNAME_EMPTY);
+                                JSMSG_BACKEND_CLIENT_CERTIFICATE_NO_CERTIFICATE);
+      return false;
+    }
```

Comment:
Isn't this dead code, as the `if (!found)` case is underneath the `if (found)` above on line 1491? Perhaps this should be reverted back to checking for an empty string, as that case would be distinct from the string being a `nullptr`.

Reply:
I believe not since `found` is reused on the stack throughout this routine.

Reply:
Ah, I missed that it was being overwritten. Thanks!